<!DOCTYPE html>
<html lang="en">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
  <meta name="description" content="">
  <meta name="author" content="">

  <title>VariantValidator - Batch Help</title>

  <!-- Bootstrap core CSS -->
  <link href="/static/vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">

  <!-- Custom styles for this template -->
  <link href="/static/css/business-frontpage.css" rel="stylesheet">

  <link href="/static/css/vvweb.css" rel="stylesheet">
  <link href="/static/cookielaw/css/cookielaw.css" rel="stylesheet">

  <script src="https://kit.fontawesome.com/ff8a38ffc2.js" crossorigin="anonymous"></script>

  <!-- Favicon -->
  <link rel="apple-touch-icon" sizes="180x180" href="/static/img/favicon/apple-touch-icon.png">
  <link rel="icon" type="image/png" sizes="32x32" href="/static/img/favicon/favicon-32x32.png">
  <link rel="icon" type="image/png" sizes="16x16" href="/static/img/favicon/favicon-16x16.png">
  <link rel="manifest" href="/static/img/favicon/site.webmanifest">

  
  

</head>

<body>

  <div class="overlay"></div>
  <!-- Navigation -->
  <nav class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top">
    <div class="container">
      <a class="navbar-brand" href="/">VariantValidator.org</a>
      <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
        <span class="fas fa-bars" style="font-size: larger"></span>
      </button>
      <div class="collapse navbar-collapse" id="navbarResponsive">
        <ul class="navbar-nav ml-auto">
          <li class="nav-item ">
            <a class="nav-link" href="/">Home</a>
          </li>
          <li class="nav-item dropdown ">
            <a class="nav-link dropdown-toggle" data-toggle="dropdown" id="navbarDropdown" role="button" aria-haspopup="true" aria-expanded="false" href="#">Tools</a>
              <div class="dropdown-menu" aria-labelledby="navbarDropdown">
                <a class="dropdown-item" href="/service/validate/">Validator</a>
                <a class="dropdown-item" href="/service/validate/batch/">Batch Validator</a>
                <a class="dropdown-item" href="/service/gene2trans/">Genes to Transcripts</a>
                <a class="dropdown-item" href="/service/vcf2hgvs/">VCF to HGVS</a>
                <a class="dropdown-item" href="https://rest.variantvalidator.org/">REST API</a>
              </div>
          </li>
          <li class="nav-item dropdown active ">
            <a class="nav-link dropdown-toggle" data-toggle="dropdown" id="navbarDropdown" role="button" aria-haspopup="true" aria-expanded="false" href="#">Information</a>
              <div class="dropdown-menu dropleft" aria-labelledby="navbarDropdown">
                <a class="dropdown-item" href="https://github.com/openvar/variantValidator/blob/master/README.md">About</a>
                <a class="dropdown-item" href="/help/contact/">Contact</a>
                <div class="dropdown-divider"></div>
                <!-- <a class="dropdown-item" href="/help/nomenclature/">Nomenclature</a> -->
                <a class="dropdown-item" href="/help/instructions/">Batch Tool</a>
                <a class="dropdown-item" href="/help/faq/" style="display: none">FAQs</a>
                <div class="dropdown-divider"></div>
                <a class="dropdown-item dropdown-toggle" id="helpSubDropdown" data-toggle="dropdown" href="#" aria-haspopup="true" aria-expanded="false">External Links</a>
                  <div class="dropdown-menu" aria-labelledby="helpSubDropdown">
                      <h6 class="dropdown-header">HGVS and UTA</h6>
                      <a class="dropdown-item" href="https://github.com/biocommons/hgvs" target="_blank">HGVS library</a>
                      <a class="dropdown-item" href="https://github.com/biocommons/uta/" target="_blank">Universal Transcript Archive</a>
                      <div class="dropdown-divider divider-header"></div>
                      <h6 class="dropdown-header">HGVS Nomenclature</h6>
                      <a class="dropdown-item" href="http://varnomen.hgvs.org/" role="button" target="_blank">HGVS/varnomen</a>
                      <div class="dropdown-divider divider-header"></div>
                      <h6 class="dropdown-header">Variant Databases</h6>
                      <a class="dropdown-item" href="http://grenada.lumc.nl/LSDB_list/lsdbs" target="_blank">Locus Specific Database list</a>
                      <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/clinvar/" target="_blank">ClinVar</a>
            		  <a class="dropdown-item" href="https://cancer.sanger.ac.uk/cosmic" target="_blank">COSMIC</a>
                      <div class="dropdown-divider divider-header"></div>
                      <h6 class="dropdown-header">Reference Sequence Portals</h6>
                      <a class="dropdown-item" href="https://www.ncbi.nlm.nih.gov/refseq/" target="_blank">NCBI RefSeq</a>
                      <a class="dropdown-item" href="http://www.lrg-sequence.org/" target="_blank">Locus Reference Genomic</a>
                      <a class="dropdown-item" href="http://www.ensembl.org/index.html" target="_blank">ENSEMBL</a>
                      <a class="dropdown-item" href="https://genome.ucsc.edu/cgi-bin/hgGateway" target="_blank">UCSC Genome Browser</a>
                      <div class="dropdown-divider divider-header"></div>
                      <h6 class="dropdown-header">Similar Sites</h6>
                      <a class="dropdown-item" href="https://mutalyzer.nl/" target="_blank">Mutalyzer</a>
                  </div>
              </div>
          </li>
          
          <li class="nav-item dropdown">
            <a class="nav-link dropdown-toggle" data-toggle="dropdown" id="navbarDropdown" role="button" aria-haspopup="true" aria-expanded="false" href="#">
              Raymond Dalgleish
            </a>
              <div class="dropdown-menu" aria-labelledby="navbarDropdown">
                <a class="dropdown-item" href="/profile/">Profile</a>
                <div class="dropdown-divider"></div>
                <a class="dropdown-item" href="#" data-toggle="modal" data-target="#logoutModal">Sign Out</a>
              </div>
          </li>
          
        </ul>
      </div>
    </div>
  </nav>

  

  <!-- Header -->
  <header class="bg-primary py-2">
    <div class="container h-25">
      <div class="row h-25 align-items-center">
        <div class="col-9">
          <h1 class="display-4 text-white mt-4 mb-2">Batch Tool Help</h1>
          <p class="lead mb-0 text-white-75">Using the VariantValidator Batch Tool to validate variant descriptions</p>
        </div>
      <div class="col-3 text-right">
        <img src="/static/img/logos/VV_logo.png" alt="VariantValidator Logo" class="img-fluid" style="max-height: 200px">
      </div>
      </div>
    </div>
  </header>
  

  <!-- Page Content -->
  <div class="container mt-5">
  
  
  <div class="row">
    <div class="col-md-12 mb-5">
      <h3>The basic steps:</h3>
      <p>
        <a href="/" title="VariantValidator Home Page">VariantValidator</a>
        is a web-based tool that allows the checking of human genome sequence variant descriptions for compliance
        with the HGVS Variant Nomenclature. Variants can be checked individually or multiple variant descriptions
        can be checked simultaneously by submitting them to the
        <a href="/service/validate/batch/" title="Batch Validator" >Batch Validator</a>.
      </p>
      <p>
        Variant descriptions are pasted into the Input Variant Descriptions data entry box with one description per
        line. Instructions are provided on-screen. Validation can be limited to a particular gene or genes. This is
        particularly useful if you have a long list of variants that are expressed in the context of genome
        coordinates, but only those that fall into a particular gene or genes are of interest. HGNC gene symbols
        are used to impose gene search limits and the input is not case-sensitive. The results are returned to you
        by email, so you must enter a valid email address. Finally, you must select a genome build (GRCh38 or GRCh37)
        to allow the validation of variant descriptions, such as 17-50198002-C-A or chr17:50198002C>A, where the
        appropriate genome build cannot be deduced from the description itself. Needless to say, it's not possible to
        simultaneously submit a mix of variant descriptions from both genome builds.
      </p>

      <h3>Formatting the data for validation</h3>
      <p>The ideal method is to use a plain text editor. Use of a word processor is not recommended because of the
        potential to carry over non-valid characters from your list of variant descriptions into the input box.
        VariantValidator has been designed to strip out invalid characters but it is better to avoid them being
        there in the first place. Copying variant descriptions directly from journal articles and from computer
        screens can be equally problematic.</p>

      <p><strong>Note:</strong> It is vitally important that there be no trailing "invisible" characters,
        such as spaces or tabs, immediately after a variant description. The presence of such characters can
        result in the submission of a blank variant description which will return a "Variant description is
        not in an accepted format" warning at the end of the batch validator output. We are aware of this problem
        and will apply a software patch as soon as possible to handle the issue.</p>

      <p>A short list of valid variant descriptions might look like this:</p>
      <pre>
<code>NM_000088.3:c.589G>T
NC_000017.10:g.48275363C>A
NG_007400.1:g.8638G>T
LRG_1:g.8638G>T
LRG_1t1:c.589G>T
17-50198002-C-A
chr17:50198002C>A</code></pre>
      <p>Note that the final two variant descriptions are in the context of GRCh38. It's also worth noting that each of these
        seven variant descriptions is valid and should return no error messages.</p>

      <p>The following variant descriptions are invalid and are each expected to return an error message:</p>
      <pre>
<code>NM_000088.3:c.589C>T
NC_000071.10:g.48275363C>A
NG_007400.2:g.8638G>T
LRG_1:g.8638G>N
LRG_1t3:c.589G>T
17-50198002-G-A
chr17:550198002C>A
COL5A1:c.5071A>T
NM_000088.3:c.589GG>CT
NM_000500.7:c.-107-19C>T</code></pre>
      <p>Plain text files containing these variant descriptions can be downloaded as
        <a href="/static/txt/valid_variant_test_set.txt" download>valid_variant_test_set.txt</a> and
        <a href="/static/txt/invalid_variant_test_set.txt" download>invalid_variant_test_set.txt</a></p>

      <h3>What happens next?</h3>
      <p>When you press the submit button, you will immediately receive confirmation on screen that your job has been
        submitted. In addition, you will receive a confirmation by email which includes a Request ID and the date
        and time at which the request was submitted. The time taken for the return of your results will depend on the
        number of other jobs ahead of you in the queue and the number of variant description to be validated in your job.
        The email confirmation explains this and provides contact information if you have any queries. When your job is
        complete, you will receive another email explaining how to download your results from our server. The download is
        initiated by clicking on the link in the email. Results are stored only for seven days and are then deleted.</p>

      <h3>What have I downloaded?</h3>
      <p>When you click on the link, the results are returned to you as plain text that you will need to save on your computer.
        Forcing you to save the file is deliberate as the results will not be stored permanently beyond seven days.
        The data are in what is known as tab-delimited format. It is human readable, but is best opened in a spreadsheet
        program such as Microsoft Excel or OpenOffice Calc. Just accept the default actions for which you will be prompted by Excel or Calc.</p>

      <p>Once your data have been imported into your spreadsheet program of choice, you can customise the view
        by adjusting the widths of the columns in the table view. No other adjustments should be necessary,
        and you should now be able to easily read the results. Not all of the columns will necessarily be of value to you.</p>

      <p>The key first task is to check for any errors or warnings in column B (validation_warnings) which
        reports both errors and warnings (Including any automatic corrections which VariantValidator may have
        performed). Errors indicate that a variant description has failed to validate correctly, but warnings are
        intended to convey information. If you encounter any errors, you should fix them and resubmit corrected
        descriptions to the Batch Validator. Please submit only those descriptions that need to actually be validated.
        Resist the temptation to resubmit the variants that have already passed validation.</p>

      <p>You must use the outputs provided by VariantValidator for submission to journals, not the variant descriptions
         that you submitted</p>

      <p>When submitting intronic variants to journals, you must use either a chromosomal context or RefSeqGene context
         intronic variant description as provided by the tool</p>

      <p>Two common warnings that you might see relate to Locus Reference Genomic (LRG) and NCBI RefSeq
        transcript and gene reference sequences. The validator automatically projects variants onto LRG
        reference sequences (gene and transcript(s)) if an LRG exists for the gene to which the variant maps.
        A warning will be generated if that LRG is currently flagged as "pending". LRG-based variant descriptions
        should only be used for LRGs that are finalised.</p>

      <p>If you have submitted a variant description for validation using RefSeq transcript or gene that
        is not the most recent version, this will trigger a warning. It's not an error. The warning is to simply
        alert you that a newer version of the transcript or gene sequence record exists and you may wish to resubmit
        the variant description in the context of that newer version. If your variant description validates correctly
        in the context of an older version of a reference sequence, that is okay. The HGVS variant nomenclature
        guidelines do not require use of the most recent version.
      </p>

      <p>Example output files, derived form the example data above, can be downloaded as
        <a href="/static/txt/valid_variant_test_results.txt" download>valid_variant_test_results.txt</a>
        and <a href="/static/txt/invalid_variant_test_results.txt" download>invalid_variant_test_results.txt</a>
      </p>
      <p>Finally, and importantly, once you have imported your results into a spreadsheet,
        be sure to save the spreadsheet in its native Excel or Calc format. If you forget, you will have to repeat the import process.</p>

      <h3>Getting help</h3>
      <p>VariantValidator has been developed to meet the many needs of our users.
        We understand that many users find it difficult to write variant descriptions for submission to VariantValidator,
        and we realise that many aspects of the HGVS nomenclature are difficult to comprehend
        (for expert and novice users alike). We encourage our users to contact us if they are having difficulty with
        reporting their sequence variants by filling in a <a href="/help/contact/">basic web form</a>.
        We will endeavour to contact you as soon as we can to provide you with the guidance that you require.
        Users can also report processing errors and make feature requests by contacting us on
        <a href="https://github.com/openvar/variantValidator/issues" target="_blank">GitHub</a>.</p>

      <h3>Metadata</h3>
      <p>VariantValidator will automatically add meta data to the top and tail of your file.
        The data will provide information such as the software version and database versions.
        We recommend that you retain these data as it will help us to resolve any issues that you may encounter in the future.</p>

      <h3>Warnings that are not errors:</h3>
      <p>Some warning messages that are produced by the batch validator are not errors, but are designed to provide
        useful guidance. Examples (in no particular order) of such warnings include:</p>
      <ul class="instructions-lst">
        <li><b>A more recent version of the selected reference sequence NM_000022.2 is available (NM_000022.3):</b><br>
          This is a specific example for RefSeq record NM_000022. Although it is considered best-practice to report
          variants in the context of the most up-to-date reference sequence, HGVS does not demand this. Hence, it is
          up to the user, or perhaps the policy of journals in which they hope to publish results, whether they
          choose to use the updated description.
        </li>
        <li><b>No transcripts found that fully overlap the described variation in the genomic sequence:</b><br>
          Some genome sequence variants will lie outside of genes and will not project onto (align with) a gene
          transcript. Some variants might project only partially onto a transcript. In both instances,
          this warning will be generated.
        </li>
        <li><b>The current status of LRG_XYZ is pending therefore changes may be made to the LRG reference sequence:</b><br>
          Locus Reference Genomic (LRG) reference sequences may be made available for use prior to the content being
          fully approved and LRG record being designated as being “Public”. Any LRG that is designated as being
          “Pending” is not stable and may be subject to change.
        </li>
        <li><b>This coding sequence variant description spans at least one intron; use of the corresponding genomic sequence variant descriptions may be invalid:</b><br>
          Sequence variants, such as deletions, that have been discovered by RNA sequencing might appear to span at
          least one intron. However, such a deletion might not be due to a single mutational event and attempts to
          project the deletion from a transcript to the genome might be invalid.
        </li>
        <li><b>Protein level variant descriptions are not fully supported due to redundancy in the genetic code:</b><br>
          VariantValidator can verify the syntax and the reference amino acid(s) specified in a protein-level variant
          description. However, it is impossible to project such a variant to a transcript or to the genome because of
          redundancy in the genetic code.
        </li>
        <li><b>RefSeqGene record not available:</b><br>
          Only some genes have an NCBI RefSeqGene record. This warning simply indicates that no RefSeqGene record
          exists for the gene in which the variant that is being validated resides.
        </li>
      </ul>

      <h3>How to cite VariantValidator</h3>
      <p>If you use VariantValidator in your research, we would be grateful if you would cite the publication that describes
        it:</p>
      <p>VariantValidator: Accurate validation, mapping and formatting of sequence variation descriptions.<br>
        Freeman PJ, Hart RK, Gretton LJ, Brookes AJ, Dalgleish R. (2018) <i>Human Mutation</i> 39:61-68.
        <a href="https://doi.org/10.1002/humu.23348" target="_blank">https://doi.org/10.1002/humu.23348</a>
      </p>
      <address class="font-italic">Raymond Dalgleish, Version 0.5, 4 June 2020</address>
    </div>
  </div>


  
  </div>

  <!-- Footer -->
  <footer class="py-4 bg-light">
    <div class="container">
      <div class="row">
        <div class="col col-lg-3">
          <a href="http://https://www.manchester.ac.uk/" target="_blank">
            <img class="img-fluid" src="/static/img/logos/Manchester_logo.png" style="min-width: 100px" alt="University of Manchester Logo">
          </a>
        </div>
        <div class="col col-lg-6">
          <p class="m-0 text-center text-dark">Copyright &copy; 2019 VariantValidator Contributors</p>
        </div>
        <div class="col col-lg-3">
          <a href="http://le.ac.uk" target="_blank">
            <img class="img-fluid" src="/static/img/logos/uniofleicesterlogo.png" style="min-width: 100px" alt="University of Leicester Logo">
          </a>
        </div>
      </div>
    </div>
    <!-- /.container -->
  </footer>

  <!-- Logout Modal -->
  <!-- Modal -->
  <div class="modal fade" id="logoutModal" tabindex="-1" role="dialog" aria-labelledby="logoutModalLabel" aria-hidden="true">
    <div class="modal-dialog" role="document">
      <div class="modal-content">
        <div class="modal-header">
          <h5 class="modal-title" id="logoutModalLabel">Sign Out</h5>
          <button type="button" class="close" data-dismiss="modal" aria-label="Close">
            <span aria-hidden="true">&times;</span>
          </button>
        </div>
        <div class="modal-body">
          <p>Are you sure you want to sign out?</p>
        </div>
        <div class="modal-footer">
          <form method="post" action="/accounts/logout/">
            <input type="hidden" name="csrfmiddlewaretoken" value="tHSOQ1ymlroKmm5KW0Lz6VRGBNTtObVNrBPwfsE7GCNYa5vhSv4dvt3Ao8suQlra">
            
            <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
            <button type="submit" class="btn btn-primary">Sign Out</button>
          </form>
        </div>
      </div>
    </div>
  </div>

  <!-- Bootstrap core JavaScript -->
  <script src="https://code.jquery.com/jquery-3.4.1.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
  <script src="https://cdnjs.cloudflare.com/ajax/libs/popper.js/1.14.7/umd/popper.min.js" integrity="sha384-UO2eT0CpHqdSJQ6hJty5KVphtPhzWj9WO1clHTMGa3JDZwrnQq4sF86dIHNDz0W1" crossorigin="anonymous"></script>
  <script src="https://stackpath.bootstrapcdn.com/bootstrap/4.3.1/js/bootstrap.min.js" integrity="sha384-JjSmVgyd0p3pXB1rRibZUAYoIIy6OrQ6VrjIEaFf/nJGzIxFDsf4x0xIM+B07jRM" crossorigin="anonymous"></script>
  <script src="/static/js/cookielaw.js"></script>
  <script src="/static/js/vvweb.js"></script>
  
  

</body>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','https://www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-77024113-1', 'auto');
  ga('send', 'pageview');
</script>
</html>